File: Pandacap.Data/Credentials/FurAffinityCredentials.fs
namespace Pandacap.Data

open System.ComponentModel.DataAnnotations
open Pandacap.FurAffinity
open Pandacap.ConfigurationObjects

/// The active credentials for Fur Affinity.
type FurAffinityCredentials() =
    [<Key>]
    member val Username = "" with get, set
    member val A = "" with get, set
    member val B = "" with get, set

    interface IFurAffinityCredentials with
        member this.A = this.A
        member this.B = this.B
        member _.UserAgent = UserAgentInformation.UserAgent

File: Pandacap.Database/FurAffinityCredentials.cs
using Pandacap.Constants;
using Pandacap.FurAffinity.Interfaces;
using Pandacap.UI.Badges;
using Pandacap.UI.Elements;
using System.ComponentModel.DataAnnotations;

namespace Pandacap.Database
{
    public class FurAffinityCredentials : IFurAffinityCredentials, IExternalCredentials
    {
        [Key]
        public string Username { get; set; } = "";

        public string A { get; set; } = "";
        public string B { get; set; } = "";

        string IFurAffinityCredentials.UserAgent =>
            UserAgentInformation.UserAgent;

        Badge IExternalCredentials.Badge => Badges.FurAffinity;

        string IExternalCredentials.PlatformName => "Fur Affinity";
    }
}
